R/plumber.R

```diff
@@ -177,10 +177,9 @@ plumber <- R6Class(
       private$notFoundHandler <- default404Handler
 
       if (!is.null(file)){
-        private$lines <- readLines(file)
-        private$parsed <- parse(file, keep.source=TRUE)
-
-        source(file, local=private$envir, echo=FALSE, keep.source=TRUE)
+        private$lines <- readUTF8(file)
+        private$parsed <- parseUTF8(file)
+        sourceUTF8(file, private$envir)
```

Comment:
I think we should get an understanding of why we're both parsing the file and sourcing it. That will help us understand whether we need to preserve the `keep.source=TRUE` behavior.

Reply:
I believe we were misled by the name of `keep.source` within `source`.  

`source` docs:
> echo: logical; if ‘TRUE’, each expression is printed after parsing, before evaluation.
> keep.source: logical: should the source formatting be retained when echoing expressions, if possible?

I believe it was used for debugging purposes when developing. (Set `echo` to true and display nicely on screen.)

--------------------

We are more concerned with `parse`'s `keep.source`.

`parse` docs:
> keep.source: a logical value; if ‘TRUE’, keep source reference information.

Which I don't know why it wouldn't be TRUE.  If errors were a big deal, users would be sanitizing them before responding.

Reply:
The extra sourcing is doe to populate local variables in the envir for the function.  Explanation: https://github.com/trestletech/plumber/issues/329

While maybe it is a bug that the code will be evaluated twice, this must be kept for desired behavior.

Reply:
This line can be removed once `activateBlock` is changed so that every expression given to it is `eval`ed.